_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -177,10 +195,12 @@ class FrameStorage
                 const UMC::FrameMemID mid = frame.ref_frame_map[ref_index];
                 if (mid >= 0)
                 {
-                    VP9_CHECK_AND_THROW((m_frameAllocator->DecreaseReference(mid) == UMC::UMC_OK), MFX_ERR_UNKNOWN);
+                    status = m_frameAllocator->DecreaseReference(mid);
                 }
             }
         }
+        status = m_frameAllocator->DecreaseReference(frame.currFrame);
+        CheckStatus(status);
```

Comment:
if `DecreaseReference` returns an error, what does it mean? It looks like a logical error, we can't do any graceful handling here, just throw an exception immediately, not at the end of method.

Reply:
We had successfully locked tons of resources in LockResources function. If some DecreaseRefernce fails we throw an exception handling the exception we have two major possibilities the first one is to turn off MediaSDK engine at all, the second one tries to proceed work form some point using existing resources (after Reset, Close/Init, etc.), in that case, we can't guarantee the locked surfaces will be propper uninitialized or freed. So my point is clean up all resources before throw exception.